src/components/geocoder-panel.js

```diff
@@ -133,97 +83,95 @@ function isValid(key) {
   return /pk\..*\..*/.test(key);
 }
 
-export class GeocoderPanel extends Component {
-  static propTypes = {
-    isGeocoderEnabled: PropTypes.bool.isRequired,
-    mapboxApiAccessToken: PropTypes.string.isRequired
-  };
-
-  state = {
-    selectedGeoItem: null
-  };
-
-  removeGeocoderDataset() {
-    this.props.dispatch(removeDataset(GEOCODER_DATASET_NAME));
-  }
+export default function GeocoderPanelFactory() {
+  class GeocoderPanel extends Component {
+    static propTypes = {
+      isGeocoderEnabled: PropTypes.bool.isRequired,
+      mapboxApiAccessToken: PropTypes.string.isRequired,
+      mapState: PropTypes.object.isRequired,
+      updateVisData: PropTypes.func.isRequired,
+      removeDataset: PropTypes.func.isRequired,
+      updateMap: PropTypes.func.isRequired,
+
+      transitionDuration: PropTypes.number,
+      width: PropTypes.number
+    };
+
+    removeGeocoderDataset() {
+      this.props.removeDataset(GEOCODER_DATASET_NAME);
+    }
 
-  onSelected = (viewport = null, geoItem) => {
-    const {
-      center: [lon, lat],
-      text,
-      bbox
-    } = geoItem;
-    this.removeGeocoderDataset();
-    this.props.dispatch(
-      addDataToMap({
-        datasets: [generateGeocoderDataset(lat, lon, text)],
-        options: {
+    onSelected = (viewport = null, geoItem) => {
+      const {
+        center: [lon, lat],
+        text,
+        bbox
+      } = geoItem;
+      this.removeGeocoderDataset();
```

Comment:
I thought we were trying to get away from it and I spent my time working on the solution to avoid creating new datasets/layers outside the mapbox. But I guess that's fine.

Reply:
this PR only applies cosmetic changes, e.g. pass actions instead of dispatch. your work on changing the rendering methods still applies